src/cli/common.rs

```diff
@@ -157,24 +161,27 @@ pub fn set_globals(verbose: bool, quiet: bool) -> Result<Cfg> {
         ..Default::default()
     });
 
-    Ok(Cfg::from_env(Arc::new(move |n: Notification<'_>| {
-        if download_tracker.borrow_mut().handle_notification(&n) {
-            return;
-        }
-        console_notifier.borrow_mut().handle(n);
-    }))?)
+    Ok(Cfg::from_env(
+        vars,
+        Arc::new(move |n: Notification<'_>| {
+            if download_tracker.borrow_mut().handle_notification(&n) {
+                return;
+            }
+            console_notifier.borrow_mut().handle(n);
+        }),
+    )?)
 }
 
-pub fn show_channel_update(
-    cfg: &Cfg,
+pub fn show_channel_update<V: VarSource + Clone>(
```

Comment:
We could get rid of a lot of this noise by boxing the VarSource in the `Cfg` instance rather than parameterising it on it.

Reply:
Also, thinking about it, that'd be faster to build too, and not be enough of a performance issue to be worth considering against IMO.

Reply:
Then when you need to pass the var source to other things, make it take a `&dyn VarSource` instead?

Reply:
thats a possibility too, yes.